tests/test_connection.py

```diff
@@ -848,62 +848,36 @@ def test_update_fee(node_factory, bitcoind):
     l2.daemon.wait_for_log('onchaind complete, forgetting peer')
 
 
-@unittest.skipIf(not DEVELOPER, "needs DEVELOPER=1")
-def test_update_all_fees(node_factory):
-    l1, l2 = node_factory.line_graph(2, fundchannel=True)
-
-    # Set all fees as positional parameters.
-    l1.rpc.dev_setfees('12345', '6789', '123')
-    l1.daemon.wait_for_log('dev-setfees: fees now 12345/6789/123')
-    l2.daemon.wait_for_log('peer updated fee to 12345')
-
-    # Call setfees with fees passed as named parameters in different order.
-    l1.rpc.dev_setfees(slow='123', normal='4567', immediate='8901')
-    l1.daemon.wait_for_log('dev-setfees: fees now 8901/4567/123')
-    l2.daemon.wait_for_log('peer updated fee to 8901')
-
-    # Set one value at a time.
-    l1.rpc.dev_setfees(slow='321')
-    l1.daemon.wait_for_log('dev-setfees: fees now 8901/4567/321')
-    l1.rpc.dev_setfees(normal='7654')
-    l1.daemon.wait_for_log('dev-setfees: fees now 8901/7654/321')
-    l1.rpc.dev_setfees(immediate='21098')
-    l1.daemon.wait_for_log('dev-setfees: fees now 21098/7654/321')
-    l2.daemon.wait_for_log('peer updated fee to 21098')
-
-    # Verify that all fees are indeed optional in setfees call.
-    l1.rpc.dev_setfees()
-    l1.daemon.wait_for_log('dev-setfees: fees now 21098/7654/321')
-
-    # This should return finish closing.
-    l1.rpc.close(l1.get_channel_scid(l2))
-
-
 @unittest.skipIf(not DEVELOPER, "needs DEVELOPER=1")
 def test_fee_limits(node_factory):
     # FIXME: Test case where opening denied.
-    l1, l2 = node_factory.line_graph(2, opts={'dev-max-fee-multiplier': 5}, fundchannel=True)
+    l1, l2 = node_factory.line_graph(2, opts={'dev-max-fee-multiplier': 5, 'may_reconnect': True}, fundchannel=True)
 
-    # L1 asks for stupid low fees
-    l1.rpc.dev_setfees(15)
+    # L1 asks for stupid low fee (will actually hit the floor of 253)
+    l1.stop()
+    l1.set_feerates((15, 15, 15), False)
+    l1.start()
```

Comment:
Seems like we'll be doing this triple of operations a few times, we could pull this into its own method to document why we restart here.